### ISSUE TITLE
feature(AC): manually set vertical vent angle on midea ac / midea ac duo

### ISSUE THREAD

**Issue by Viss** [wuwentao/midea_ac_lan#79]
### 设备类型及型号 (或SN)

midea ac duo

### 新功能的描述

This is a continuation of this issue from last year: https://github.com/georgezhao2010/midea_ac_lan/issues/243

in the midea app, it is possible to manually set the angle of the vertical vent on the midea 12k btu ac unit and the midea duo (i have both). I would love it if it was possible to manually set the vent direction in home assistant, as I am currently having to do it maually using the midea app. 

happy to help troubleshoot or debug or run test code!

**Comment by rokam** [comment]
Can you please provide the device serial number?

`midealocal discover` should show that number.

**Comment by rokam** [comment]
I need the serial number to get the lua protocol file. This can be archived with `midealocal discover -d` command

**Comment by dcflachs** [comment]
This is great!
The new "Airflow Vertical" seems to work just like the like the control in the official smarthome app. The 5 directional options move the vent to the proper angle, I am not really certain what the "Off" option is supposed to do the AC beeps when it is selected but I dont observe anything changing on the AC. There is also some strange behavior with the control display in HA. No matter what i change the setting to it seems that the display will eventually change back to "Up" even though the AC vent does not change position. The "Logbook" seems to show events indicating the device swapping back and forth between the actual position and the UP position and occasionally the control will momentarily swap to the correct current position.

I tried out the "Airflow Horizontal" control as well, not that i expected it to do anything as my AC does not have horizontal swing. When the horizontal control is changed the AC beeps but nothing happens, again not that i expect anything would. 

I recorded a debug log as you asked but if there are any specific actions you want me to take with the logging enabled I can do that as well.

[home-assistant_midea_ac_lan_2025-08-05T01-54-17.117Z.log](https://github.com/user-attachments/files/21588460/home-assistant_midea_ac_lan_2025-08-05T01-54-17.117Z.log)

**Comment by Viss** [comment]
I havent tried this update yet, I'm waiting for it to go to HACS proper. Also holding out hope that we'll be able to control the beeps as well so we can just keep those off. Its literally the first thing I do when the unit comes on - shut it up so it stops beeping every time i issue a command.

**Comment by wuwentao** [comment]
> This is great! The new "Airflow Vertical" seems to work just like the like the control in the official smarthome app. The 5 directional options move the vent to the proper angle, I am not really certain what the "Off" option is supposed to do the AC beeps when it is selected but I dont observe anything changing on the AC. There is also some strange behavior with the control display in HA. No matter what i change the setting to it seems that the display will eventually change back to "Up" even though the AC vent does not change position. The "Logbook" seems to show events indicating the device swapping back and forth between the actual position and the UP position and occasionally the control will momentarily swap to the correct current position.
> 
> I tried out the "Airflow Horizontal" control as well, not that i expected it to do anything as my AC does not have horizontal swing. When the horizontal control is changed the AC beeps but nothing happens, again not that i expect anything would.
> 
> I recorded a debug log as you asked but if there are any specific actions you want me to take with the logging enabled I can do that as well.
> 
> [home-assistant_midea_ac_lan_2025-08-05T01-54-17.117Z.log](https://github.com/user-attachments/files/21588460/home-assistant_midea_ac_lan_2025-08-05T01-54-17.117Z.log)

thank you very much for your test result and debug log.
the off option is  a status and we can't control it to off, anyway, as official app also not support it, but it still have a off option that not select all the positions, in HA we can't support it, so I use this new off option for it.

if your devie not support airflow Horizontal, you can disable this option in your device config.

the screen display result may update by device status refresh and not by this control action.

**Comment by wuwentao** [comment]
> I havent tried this update yet, I'm waiting for it to go to HACS proper. Also holding out hope that we'll be able to control the beeps as well so we can just keep those off. Its literally the first thing I do when the unit comes on - shut it up so it stops beeping every time i issue a command.

you can enable/disable prompt tone, and it should be a split options/switch, what's your current question or issue？
do you mean it can't works for your device?

**Comment by dcflachs** [comment]
> thank you very much for your test result and debug log. 

No problem glad to help.

>the off option is a status and we can't control it to off, anyway, as official app also not support it, but it still have a off option that not select all the positions, in HA we can't support it, so I use this new off option for it.

That makes perfect sense.

> if your devie not support airflow Horizontal, you can disable this option in your device config.

I already did thanks. I just wanted to indicate that i had tried it in case it is visible in the debug log and you wondered why it was there when my device doesnt support it. 

> the screen display result may update by device status refresh and not by this control action.

Right but it does seem like the periodic status update is only intermittently seeing the correct device state for that control. I just wanted to report that as a possible issue.

Thank you for your continued work on this project.

**Comment by Viss** [comment]
ah! I found it! It looks like for some reason HA called the entity "none", so it was a little hard to find, but i got it working and now its functioning in automations!

**Comment by Viss** [comment]
Yeah i get the same thing. It visually changes back to 'Up' but it doesnt actually move back to up. I have also noticed that when trying to use the dropdown to select a vent position manually it takes a few tries for it to happen

**Comment by wuwentao** [comment]
@dcflachs thank you very much for your new issue and support. I will check the detail and try to find a solution. and will works on new issue.

@Viss @dcflachs 
in addition, for the None display issue, it's a known issue, please check my PR detail: https://github.com/wuwentao/midea_ac_lan/pull/663
I have add some comments with it. but I have missed to answer it with this info, sorry.
